### ISSUE TITLE
unable to push to private repo in hub.docker.com

### ISSUE THREAD

**Issue by kbroughton** [kubernetes/kompose#910]
$ kompose  --file docker-compose-out.yml -v up
INFO Build key detected. Attempting to build and push image 'haystackabi/hsabi-web-app:4ca21a16'
DEBU Compose file dir: /Users/kesten/projects/core
INFO Building image 'myorg/myimage:4ca21a16' from directory 'myimage'
DEBU Created temporary file /tmp/kompose-image-build-328321841 for Docker image tarballing
INFO Image 'myorg/myimage' from directory 'myimage' built successfully
DEBU Docker Compose version: 3.0
FATA Error while deploying application: k.Transform failed: Unable to push Docker image for service haystack-web-app: Unable to retrieve .docker/config.json authentication details. Check that 'docker login' works successfully on the command line.: Failed to read authentication from dockercfg

$ kompose version
1.7.0 (767ab4b)

push using docker works fine
$ docker push haystackabi/hsabi-web-app:4ca21a16
The push refers to a repository [docker.io/haystackabi/hsabi-web-app]
2a0a82b8c76c: Layer already exists
5c95d741d318: Pushing [=================>                                 ]    112MB/313MB
3f34fa06d2a5: Pushing [===================

**Comment by cdrage** [comment]
@kbroughton What OS are you using? Seems it can't find `~/.docker/config.json`

**Comment by kbroughton** [comment]
osx and targeting gke.

I commented out the build step from docker-compose-out.yml and then hit this: 
FATA Error while deploying application: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

which was resolved by
kestens-MacBook-Pro:core kesten$  gcloud auth application-default login

And kube can't read from my dockerhub private repo so i get image pull failures.
